spec/enumerable_spec.rb

```diff
@@ -0,0 +1,340 @@
+require 'enumerable'
+
+describe Enumerable do
+  array = [1, 2, 3, 4, 5, 6, 7, 8]
+  string_array = %w[uzair emanuel dan vic Oni]
+  hash = { one: 1, two: 2, three: 3 }
+
+  describe '#my_each' do
+    context 'If no block given' do
+      it 'returns enum' do
+        expect(array.my_each).to be_an Enumerator
+      end
+    end
```

Comment:
- [ ] This is nice, but you should also test for negative scenarios.

Reply:
Hello,

I am very confused. Can you please give an example where:

```
context "If no block given" do
     it "returns enum" do
        expect(array.my_each).to be_an Enumerator
      end
    end
```

In this case, the negative scenario for each statement is that if there is nothing you are passing on the function eg: `#my_each`, It will always return an enumerator. As per the Ruby API documentation.

For each of the functions in my enumerable module, I have tried to test them for arrays, string_arrays, and hashes.
Some of the functions must have a negative scenario, I understand it for example I can think of some negative scenarios for `#my_any?`, `#my_all?` and `#my_none?` and I will submit after implementing them. 

As for the `#my_each` function, I cannot think of any other negative scenario where the `if a block is not given` `#my_each` function should perform differently.

Waiting for your response!

Happy Coding!

Thanks,
Uzair Ali

Reply:
Hell @uzairali19 hope you are fine, I get your point. I understand that my_each(each) function should iterate through an array and will return each array element so that we can work or perform a task, so in this case, something like this can work.
 describe '#my_each' do
    it 'Performs a block of code if given and returns the passed enumerable' do
      expect(my_array.my_each { |x| 5 * x }).to eq(my_array)
    end
    it 'Performs a block of code if given and returns the passed enumerable' do
      expect(my_array.my_each { |x| 5 * x }).to_not eq([1, 2])
    end
  end

Reply:
Hello,

Thank you for the clarification.

Regards,
Uzair Ali